worker/pkg/workflows/datasync/workflow/workflow.go

```diff
@@ -419,6 +409,44 @@ func execRunJobHooksByTiming(ctx workflow.Context, req *jobhooks_by_timing_activ
 	return nil
 }
 
+func runSchemaInitWorkflowByDestination(
+	ctx workflow.Context,
+	logger log.Logger,
+	accountId, jobId, jobRunId string,
+	destinationIds []string,
+) error {
+	initSchemaActivityOptions := &workflow.ActivityOptions{
+		StartToCloseTimeout: 5 * time.Minute,
+		RetryPolicy: &temporal.RetryPolicy{
+			MaximumAttempts: 1,
+		},
+		HeartbeatTimeout: 1 * time.Minute,
+	}
+	for _, destinationId := range destinationIds {
+		logger.Info("scheduling Schema Initialization workflow for execution.", "destinationId", destinationId)
+		siWf := &schemainit_workflow.Workflow{}
+		var wfResult schemainit_workflow.SchemaInitResponse
+		err := workflow.ExecuteChildWorkflow(workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+			WorkflowID:    workflow_shared.BuildChildWorkflowId(jobRunId, "init-schema", workflow.Now(ctx)),
```

Comment:
worth adding the destination id into the workflow id so that it is unique. yea we are using the time here but it might still cause dupes if it happens quick enough.